### PR TITLE
debug CommonDataModel bug

### LINKED ISSUE #765: Fix discrepancy in CommonDataModel package version errors
CommonDataModel 0.3.6 is throwing following error when `coupler_driver.jl` is run: 
`ERROR: LoadError: NetCDF error: NetCDF: Invalid argument (NetCDF error code: -36)`
```
Stacktrace:
  [1] check
    @ ~/.julia/packages/NCDatasets/JrRxr/src/errorhandling.jl:25 [inlined]
  [2] nc_def_var_chunking(ncid::Int32, varid::Int32, storage::Symbol, chunksizes::Vector{Int64})
    @ NCDatasets ~/.julia/packages/NCDatasets/JrRxr/src/netcdf_c.jl:1096
  [3] chunking(v::NCDatasets.Variable{Float32, 3, NCDatasets.NCDataset{Nothing, Missing}}, storage::Symbol, chunksizes::Tuple{Int64, Int64, Int64})
    @ NCDatasets ~/.julia/packages/NCDatasets/JrRxr/src/variable.jl:232
  [4] chunking(v::CommonDataModel.CFVariable{Union{…}, 3, NCDatasets.Variable{…}, CommonDataModel.Attributes{…}, @NamedTuple{…}}, storage::Symbol, chunksize::Tuple{Int64, Int64, Int64})
    @ CommonDataModel ~/.julia/packages/CommonDataModel/G3moc/src/cfvariable.jl:551
  [5] defVar(dest::NCDatasets.NCDataset{Nothing, Missing}, varname::String, srcvar::CommonDataModel.SubVariable{Union{…}, 3, CommonDataModel.CFVariable{…}, Tuple{…}, CommonDataModel.Attributes{…}, CommonDataModel.SubVariable{…}}; kwargs::@Kwargs{})
    @ CommonDataModel ~/.julia/packages/CommonDataModel/G3moc/src/variable.jl:238
  [6] defVar
    @ ~/.julia/packages/CommonDataModel/G3moc/src/variable.jl:206 [inlined]
  [7] #defVar#13
    @ ~/.julia/packages/CommonDataModel/G3moc/src/variable.jl:266 [inlined]
  [8] defVar(dest::NCDatasets.NCDataset{Nothing, Missing}, srcvar::CommonDataModel.SubVariable{Union{…}, 3, CommonDataModel.CFVariable{…}, Tuple{…}, CommonDataModel.Attributes{…}, CommonDataModel.SubVariable{…}})
    @ CommonDataModel ~/.julia/packages/CommonDataModel/G3moc/src/variable.jl:265
  [9] write(dest::NCDatasets.NCDataset{…}, src::CommonDataModel.SubDataset{…}; include::Vector{…}, exclude::Vector{…}, _ignore_checksum::Bool)
    @ CommonDataModel ~/.julia/packages/CommonDataModel/G3moc/src/dataset.jl:354
 [10] write(dest::NCDatasets.NCDataset{Nothing, Missing}, src::CommonDataModel.SubDataset{NCDatasets.NCDataset{Nothing, Missing}, @NamedTuple{time::UnitRange{…}}, CommonDataModel.Attributes{NCDatasets.NCDataset{…}}, OrderedCollections.OrderedDict{Any, Any}})
    @ CommonDataModel ~/.julia/packages/CommonDataModel/G3moc/src/dataset.jl:316
 [11] truncate_dataset(datafile::String, name::String, datapath_trunc::String, date0::Dates.DateTime, time_start::Float64, time_end::Float64, comms_ctx::ClimaComms.SingletonCommsContext{ClimaComms.CPUSingleThreaded})
    @ ClimaCoupler.Regridder ~/CliMA/ClimaCoupler.jl/src/Regridder.jl:666
 [12] artifact_data(datapath_full::String, name::String, datapath_trunc::String, date0::Dates.DateTime, time_start::Float64, time_end::Float64, comms_ctx::ClimaComms.SingletonCommsContext{ClimaComms.CPUSingleThreaded})
    @ Main ~/CliMA/ClimaCoupler.jl/artifacts/artifact_funcs.jl:80
 [13] top-level scope
    @ ~/CliMA/ClimaCoupler.jl/experiments/AMIP/coupler_driver.jl:171
 [14] include(fname::String)
    @ Base.MainInclude ./client.jl:489
 [15] top-level scope
    @ REPL[1]:1
```

To recreate: run coupler_driver.jl with CommonDataModel @0.3.6

Current solution: Pin CommonDataModel version to 0.3.5

### PR DESCRIPTION
**Pull request by juliasloan25**
## Purpose 
closes #765 

I'm not exactly sure what caused the bug, but in v0.3.6 CommonDataModel's extension of `Base.write` was changed [here](https://github.com/JuliaGeo/CommonDataModel.jl/compare/v0.3.5...v0.3.6#diff-4123c3f14d4a74a8fd32785a9b3f65d0cb1ea2ada4137952345f0751bbdcbfeaR354). I'm not sure what about this change caused the behavior change, but changing the dataset creation to avoid using `write` works fine.

Note: buildkite is failing because of a bug in RRTMGP that was fixed in v0.13.2. https://github.com/CliMA/ClimaCoupler.jl/pull/738 updates to use that version, so after that's. merged in I'll rebase and merge this PR.

## Content
- [x] re-write dataset truncation to avoid calling NCDatasets.write, which produces a bug as of CommonDataModel v0.3.6
- [x] rename `time_start/time_end` -> `t_start/t_end` for more consistency throughout the code and easier REPL debugging

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/Regridder.jl [hunk 1]
```diff
@@ -638,33 +638,59 @@ function cgll2latlonz(field; DIR = "cgll2latlonz_dir", nlat = 360, nlon = 720, c
...
     datafile_truncated = joinpath(datapath_trunc, file_name)
 
     if ClimaComms.iamroot(comms_ctx)
         ds = NCDatasets.NCDataset(datafile, "r")
         dates = ds["time"][:]
 
+        # Find the bounding indices of the dates we need
         (start_id, end_id) = find_idx_bounding_dates(dates, date_start, date_end)
 
+        var_truncated = NCDatasets.nomissing(ds[varname][:, :, start_id:end_id])
```
Comment (LenkaNovak) [inline src/Regridder.jl:668]: This is assuming a certain number of dimensions. Could this be generalized so it always indexes into time, e.g. even for 4D arrays?

**Review by LenkaNovak — APPROVED** [review]
LGTM, @juliasloan25. Just a quick comment on how to generalize the truncation to other than 3D datasets.

**Review by anastasia-popova — APPROVED** [review]
This looks good! Thank you for tackling this

File: src/Regridder.jl (hunk 1 above)
Comment (juliasloan25) [reply src/Regridder.jl]: Good point! I tried using `view` and selecting for the time instead, and it works locally

File: test/regridder_tests.jl [hunk 2]
```diff
@@ -14,338 +14,338 @@ REGRID_DIR = @isdefined(REGRID_DIR) ? REGRID_DIR : joinpath("", "regrid_tmp/")
...
-            # Plots.plot(T_cgll) # using ClimaCorePlots
-            # Plots.contourf(Array(T_rll)[:,1])
-
-            # Delete testing directory and files
-            rm(REGRID_DIR; recursive = true, force = true)
-        end
-    end
-end
-# test dataset truncation 
+# struct TestSurfaceSimulationA <: Interfacer.SurfaceModelSimulation end
```
[the remark this answers is not in the document]
Comment (juliasloan25) [reply test/regridder_tests.jl]: Thank you!!

**Pull request merged**